### DISCUSSION TITLE
Backup all files from device

### DISCUSSION THREAD

**Discussion by adrianblakey** [bulletmark/mpr#5] [Announcements]
Please tell me the command to recursively backup (get) all files from the Pi Pico to the local directory on MacOS.

I have tried various combinations of:

```
mpr get : .

mpr get * .

mpr get /// .

mpr get * .
```

**Comment by bulletmark** [comment]
Note that `mpr` is mostly just a wrapper for `mpremote`. When you type `mpr put *.py /` (or `mpremote cp *.py :`) it is not the program which is expanding that wildcard `*`, it is the command line environment (i.e. the shell on Linux/Mac). So to copy wildcards the other way (from device to host) you would need some kind of remote wildcard extension in MicroPython/mpremote which it does not offer. So there is currently no way to do what you want unfortunately.
